Core/SubterfugeCore/Core/Players/PlayerCurrency.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using SubterfugeCore.Core.Players;
+
+namespace SubterfugeCore.Core.Players
+{
+	public class PlayerCurrency
+	{
+		/// <summary>
+		/// A table of all the player's currencies
+		/// </summary>
+		private Dictionary<String, Dictionary<String,Object>> Currencies = new Dictionary<String, Dictionary<String, Object>>();
```

Comment:
This second dictionary is `Dictionary<String,Object>` and each key in this dictionary relates to the currency itself.

Could we turn this class into something called like `CurrencyBank` or `Bank` or something, which maps `Dictionary<CurrencyType, Currency>`? Then we'd make this part:

```
["Amount"] = currencyValue,
["CanBeNegative"] = canBeNegative
```

Moved into a `Currency` class and have an enum of `CurrencyType` that we can fetch the currency with?

Reply:
I finally decided to come back and finish this since I already started it. I believe I have almost completed the suggested changes (I added a new class called Currency that stores the amount and CanBeNegative). I have 2 questions though:

1. How do I add the changes to the pull request
2. Currently, the keys are strings in order for any string to be set as a name for a value, and so values can be created whenever needed without them having to be specified before running. I am guessing that I should change this so the currencies have to be specified within an enum first (If my understanding of enums is correct)?

Reply:
> How do I add the changes to the pull request

Just commit and push to the same branch that this MR is open for. This branch says it is merging `ilost7489:master` so if you commit your new changes and push to your master branch they will show up here :)

> I am guessing that I should change this so the currencies have to be specified within an enum first

Yep, that's the plan. I figure we'd have a set number of currencies (if we have more than one) so using an Enum just helps us to not fat finger the currency name